HerPublicWebsite.BusinessLogic/Services/CsvFileCreator/CsvFileCreator.cs

```diff
@@ -45,6 +46,24 @@ public MemoryStream CreateReferralRequestFollowUpFileData(IEnumerable<ReferralRe
             );
         return GenerateCsvMemoryStreamFromFileRows(rows);
     }
+    
+    public MemoryStream CreateConsortiumReferralRequestFollowUpFileData(IEnumerable<ReferralRequest> referralRequests)
+    {
+        var rows = referralRequests
+            .GroupBy(rr => rr.CustodianCode)
+            .GroupBy(groupingByLa =>
+                LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[groupingByLa.Key].Consortium)
+            .SelectMany(groupingByConsortium =>
+                {
+                    var consortiumReferrals = groupingByConsortium.SelectMany(g => g).ToList();
+                    var consortiumStatistics = new ConsortiumStatistics(consortiumReferrals);
+                    var consortiumStatisticsRows = groupingByConsortium.Select(groupingByLa => new CsvRowConsortiumDownloadInformation(groupingByLa, consortiumStatistics));
+                    return consortiumStatisticsRows.GroupBy(laRow => laRow.Consortium, (key,consortiumDownloadInformationRows) => 
+                        consortiumDownloadInformationRows.OrderBy(consortiumDownloadInformation => consortiumDownloadInformation.Consortium).First());
```

Comment:
This feels overly-complicated. We're calculating the consortium statistics once (good), but then constructing a row for each LA in the consortium and arbitrarily picking the first one because they're all the same anyway (I don't think your `OrderBy` is doing anything here, incidentally). Feels like we should be able to skip a few steps here:
1. You're only using `requestGroupingByCustodianCode` in the `CsvRowConsortiumDownloadInformation` constructor to get the consortium name, and you already have access to that here as the group key, so may as well pass that in directly.
2. Then you can forget about the individual LAs and do everything once, returning a single `CsvRowConsortiumDownloadInformation` object per grouping.

Reply:
I've made some changes along these lines. I think I've made them correctly, but please let me know if I've missed something